### PR TITLE
chore: Move utils

### PR DESCRIPTION
**Pull request by marikaner**
## Context

AI/gen-ai-hub-sdk-js-backlog#100.

Moving it out as it will be shared between later separated packages.

### CODE REVIEW

**Comment by marikaner** [comment]
@shibeshduw can you point out to me how you ignored the checked in `index.js` file?

File: packages/ai-core/src/internal.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './utils/index.js';
```
Comment (deekshas8) [inline packages/ai-core/src/internal.ts:1]: [q] If I remember correctly, everything had to be exported from internal.ts? Or did we change the script?

Comment (marikaner) [reply packages/ai-core/src/internal.ts:2]: but in the other direction: we will allow to export whatever you want. 😊 @shibeshduw if you do that can you make sure to remove my ugly workarounds 😬

File: packages/ai-core/src/utils/deployment-resolver.test.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import nock from 'nock';
-import { AiDeployment } from '@sap-ai-sdk/ai-core';
 import {
   mockClientCredentialsGrantCall,
   aiCoreDestination,
   mockDeploymentsList
 } from '../../../../test-util/mock-http.js';
+import { AiDeployment } from '../client/AI_CORE_API/index.js';
```
Comment (marikaner) [inline packages/ai-core/src/utils/deployment-resolver.test.ts:7]:
```suggestion
import { type AiDeployment } from '../client/AI_CORE_API/index.js';
```

File: packages/ai-core/tsconfig.cjs.json [hunk 3]
```diff
@@ -2,6 +2,7 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "module": "CommonJS",
-    "outDir": "./dist-cjs"
+    "outDir": "./dist-cjs",
+    "tsBuildInfoFile": "./dist-cjs/.tsbuildinfo"
```
Comment (marikaner) [inline packages/ai-core/tsconfig.cjs.json:6]:
```suggestion
    "outDir": "./dist-cjs"
```

File: packages/gen-ai-hub/src/orchestration/orchestration-client.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ import { OrchestrationModuleConfig, Prompt } from './orchestration-types.js';
  * Get the orchestration client.
  */
 export class OrchestrationClient {
+  // TODO: document constructor
```
Comment (marikaner) [inline packages/gen-ai-hub/src/orchestration/orchestration-client.ts:16]:
```suggestion
```

**Review by shibeshduw — APPROVED** [review]
LGTM.

**Pull request merged**